src/WebJobs.Script.WebHost/Helpers/TableStorageHelpers.cs

```diff
@@ -74,38 +73,49 @@ internal static void QueueBackgroundTablePurge(CloudTable currentTable, CloudTab
             });
         }
 
-        internal static async Task DeleteOldTablesAsync(CloudTable currentTable, CloudTableClient tableClient, string tableNamePrefix, ILogger logger)
+        internal static async Task DeleteOldTablesAsync(TableClient currentTable, TableServiceClient tableClient, string tableNamePrefix, ILogger logger)
         {
             var tablesToDelete = await ListOldTablesAsync(currentTable, tableClient, tableNamePrefix);
             logger.LogDebug($"Deleting {tablesToDelete.Count()} old tables.");
             foreach (var table in tablesToDelete)
             {
                 logger.LogDebug($"Deleting table '{table.Name}'");
-                await table.DeleteIfExistsAsync();
+                await tableClient.DeleteTableAsync(table.Name);
```

Comment:
Will this throw if it does not exist? Do we need a catch statement or is there a no-throw overload?

Reply:
This is the only overload that I've seen. On my tests If it doesn't exist it returns a 404 as part of the response, but it's not throwing.

Reply:
So long as it does not throw.